cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -208,8 +208,14 @@ public void testServiceManager() throws Exception {
     Assert.assertEquals(0, history.size());
 
     // requesting with either RUNNING or ALL will return one record
+    Tasks.waitFor(1, new Callable<Integer>() {
+      @Override
+      public Integer call() throws Exception {
+        return countService.getHistory(ProgramRunStatus.RUNNING).size();
+      }
+    }, 5, TimeUnit.SECONDS, 10, TimeUnit.MILLISECONDS);
```

Comment:
I think the default value of 50 ms, instead of the 10ms you provided, is better.
Just remove the last two params and it'll use the default value of 50ms.
Remainder LGTM.